### PR TITLE
Upgrade to Firebase Crashlytics from Firebase Crash Reporting

### LINKED ISSUE #1040: Use Crashlytics instead of Firebase Crash
From the [Firebase Crash docs](https://firebase.google.com/docs/crash/): "Crashlytics will soon become the primary crash reporter for Firebase." @nribeka @shobhitagarwal1612 @yanokwa I was totally wrong when I predicted Google would double down on Firebase Crash!! It looks like they're merging them instead.

The [Crashlytics getting started guide](https://fabric.io/kits/android/crashlytics/summary?ref=fb) currently says "Fabric and Firebase have joined forces! Soon, Crashlytics will become the primary crash reporter for Firebase. As we make this a more seamless experience, you can get Crashlytics by creating a Fabric account and installing the Crashlytics SDK." 

I think we should wait a bit for this "more seamless experience" so I've tagged this for a future release. Please comment here if you hear anything new about this.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1040 

#### What has been done to verify that this works as intended?
I didn't test it because we need to enable Crashlytics (I don't have permissions).
I used this guide https://firebase.google.com/docs/crashlytics/upgrade-from-crash-reporting?authuser=0
Once we enable Crashlytics and merge this pr we can release a beta version and investigate logs.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,11 +3,13 @@ buildscript {
     repositories {
         jcenter()
         google()
+        maven { url 'https://maven.fabric.io/public' }
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.0.1'
-        classpath 'com.google.gms:google-services:3.0.0'
+        classpath 'com.google.gms:google-services:3.1.2'
```
Comment (lognaturel) [inline build.gradle]: I think you can go straight to 3.2.0

**Review by lognaturel — COMMENTED** [review]
I've tried producing a crash with the instructions at https://firebase.google.com/docs/crashlytics/force-a-crash and it has not been registered.

Odd log lines I see:
```
D/FirebaseApp(11280): com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
E/Fabric  (11280): Failed to retrieve settings from https://settings.crashlytics.com/spi/v2/platforms/android/apps/org.odk.collect.android/settings
W/System.err(11280): remove failed: ENOENT (No such file or directory) : /data/data/org.odk.collect.android/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/log-files/crashlytics-userlog-5AAC51730339-0001-2C10-2F5572E257FB.temp
```

So far no insights as to what is going wrong. @grzesiek2010 you should have full access to try it for yourself.

**Comment by shobhitagarwal1612** [comment]
I remember adding an API key in manifest while configuring crashlytics. Is that what is missing?

**Comment by lognaturel** [comment]
I’m not seeing anything about API keys anywhere!

**Comment by shobhitagarwal1612** [comment]
https://fabric.io/kits/android/crashlytics/install

**Comment by lognaturel** [comment]
We want to use Firebase Crashlytics and not go through Fabric at all. https://firebase.google.com/docs/crashlytics/ - if you look at the setup instructions, it explicitly says to remove any existing Fabric API key.

**Comment by grzesiek2010** [comment]
Ok, I was able to finish the configuration. Now it should work well.

**Comment by lognaturel** [comment]
Fantastic! So mysterious that the debug and release variants didn't do what was needed.

One quick unused resource to take care of and then we can merge.

**Comment by grzesiek2010** [comment]
It was a problem with lint and the second with jacco + fabric. it's fixed.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thanks, @grzesiek2010! 🎉

**Pull request merged**